Generate workspace from a build script

xelxebar opened the pull request:
Manually curating code_dump.txt was annoying and I had been meaning to do this:

    Usage: ./build-workspace.apl [workspace file]

where <workspace file> defaults to `APL.dws'. Does the script also work on windows?

Commit:
Generate workspace from a build script

Instead of manually curating a list of function definitions in code_dump.txt
this commit opts to generate the workspace from the text sources themself.

    Usage: ./build-workspace.apl [workspace file]

where <workspace file> defaults to `APL.dws'.

abrudz commented on build-workspace.apl:
```diff
@@ -29,30 +38,32 @@ chkrange ← {⍺←1 ⋄ 1<⍴⍴⍵:0 perr 4 ⋄ >/w←2⍴⍵:0 perr 5 ⋄ ~
+     {}{⍞}⍣{⍺≡,'.':1 ⋄ 0⊣z,←⊂⍺}⍬
+   ∇
+
+   ∇ m←{0≠c←chkline ⍵:_←c ⋄ 0≠c←chkrange⊢x y←2⍴⍺:_←c ⋄ (x≤⍵)∧(⍵≤y):0 perr 11
+     z←(1+y-x)↑(x-1)↓∆ ⋄ 0≠c←d x y:_←c ⋄ ∆∆∘←∆ ⋄ ∆f∘←1 ⋄ __∘←_ ⋄ _∘←⍵+(⍵<x)×≢z ⋄ ∆∘←∆(↑,z,↓)⍨_-≢z ⋄ _←0}
+   ∇
```
You shouldn't have `∇`s around dfn definitions in shell scripts.

abrudz commented on build-workspace.apl:
```diff
@@ -1,4 +1,13 @@
-﻿       E ← {∆f∘←0⊣of←∆f ⋄ 0≠e ⍵:∆f∘←of ⋄ _←0}
+#!/usr/bin/env dyalogscript
```
Yse .apls for shellscripts; will work under Windows too.

Commit:
Definition of `m` is a dfn

Apparently, the behaviour of `∇` just puts the interpreter in
"function definition mode", which means that `∇ f←{...}` simply
defines a dfn, since it doesn't parse as a tradfn header.

Commit:
File extension for dyalogscript is apls

This is for Windows compatibility.

Commit:
build-workspace.apls: Reproducibly save

By default, or with left argument 1, `⎕SAVE` resumes execution right
after the call to `⎕SAVE`. This implicitly causes State Indicator
information to also be saved, which apparently embeds non-deterministic
data into the workspace.

arcfide commented:
Heh, this seems like a neat thing to try, so let's do it. :-)
